darts/tests/dataprocessing/transformers/test_midas.py

```diff
@@ -0,0 +1,460 @@
+        For example, there are always exactly three months in a quarter, but the number of days in a month differs.
+        So the monthly to quarterly transformation is possible, while the daily to monthly MIDAS transform is
+        impossible.
+        """
+        daily_times = pd.date_range(start="01-2020", end="09-30-2020", freq="D")
+        daily_values = np.arange(1, len(daily_times) + 1)
+        daily_ts = TimeSeries.from_times_and_values(
+            times=daily_times, values=daily_values, columns=["values"]
+        )
+
+        midas = MIDAS(low_freq="M")
+        self.assertRaises(ValueError, midas.fit_transform, daily_ts)
+
+    def test_inverse_transform_prediction(self):
+        """
+        Check that inverse-transforming the prediction of a model generate the correct time index when
+        using frequency anchored either at the start or the end of the quarter.
+        """
+        # low frequency : QuarterStart
+        monthly_ts = TimeSeries.from_times_and_values(
+            times=pd.date_range(start="01-2020", periods=24, freq="M"),
+            values=np.arange(0, 24),
+            columns=["values"],
+        )
+        monthly_train_ts, monthly_test_ts = monthly_ts.split_after(0.75)
+
+        model = LinearRegressionModel(lags=2)
+
+        midas_quarterly = MIDAS(low_freq="QS")
+        # shape : [6 quarters, 3 months, 1 sample]
+        quarterly_train_ts = midas_quarterly.fit_transform(monthly_train_ts)
+        # shape : [2 quarters, 3 months, 1 sample]
+        quarterly_test_ts = midas_quarterly.transform(monthly_test_ts)
+
+        model.fit(quarterly_train_ts)
+
+        # 2 quarters = 6 months forecast
+        pred_quarterly = model.predict(2)
+        pred_monthly = midas_quarterly.inverse_transform(pred_quarterly)
+        # verify prediction time index in both frequencies
+        self.assertTrue(pred_quarterly.time_index.equals(quarterly_test_ts.time_index))
+        self.assertTrue(pred_monthly.time_index.equals(monthly_test_ts.time_index))
+
+        # "Q" = QuarterEnd, the 2 "hidden" months must be retrieved
+        midas_quarterly = MIDAS(low_freq="Q")
+        quarterly_train_ts = midas_quarterly.fit_transform(monthly_train_ts)
+        quarterly_test_ts = midas_quarterly.transform(monthly_test_ts)
+
+        model.fit(quarterly_train_ts)
+
+        pred_quarterly = model.predict(2)
+        pred_monthly = midas_quarterly.inverse_transform(pred_quarterly)
+        # verify prediction time index in both frequencies
+        self.assertTrue(pred_quarterly.time_index.equals(quarterly_test_ts.time_index))
+        self.assertTrue(pred_monthly.time_index.equals(monthly_test_ts.time_index))
+
+    def test_multiple_ts(self):
+        """
+        Verify that MIDAS works as expected with multiple series of different "high" frequencies.
+        """
+        quarterly_univariate_ts = TimeSeries.from_times_and_values(
+            times=pd.date_range(start="2000-01-01", periods=12, freq="Q"),
```

Comment:
Could we use a monthly series and a quarterly series? then we can test the support for multiple ts with different high_freq to the same low freq.

Reply:
This test is already fitting both monthly (`self.monthly_ts`) and quarterly to yearly, should it test an additional scenario?